tests/test_actions.py

```diff
@@ -640,3 +654,181 @@ def test_describe(self) -> None:
+    @beat("{} tries to Action3")
+    def perform_as(self, actor: Actor):
+        actor.will(Quietly(Action4()))
+
+
+class Action4(Performable):
+    @beat("{} tries to Action4")
+    def perform_as(self, actor: Actor):
+        actor.will(Quietly(See(SimpleQuestion(), IsEqualTo(True))))
+
+
+class TestQuietlyDebug:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        settings.DEBUG_QUIETLY = False
+        yield
+        settings.DEBUG_QUIETLY = False
+
+    def test_kinking(self, Tester: Actor, mocker: MockerFixture):
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Tester.will(Quietly(FakeAction()))
+
+        assert mock_kink.call_count == 1
+        assert mock_clear.call_count == 2
+        assert mock_flush.call_count == 1
+
+    def test_skip_creation(self) -> None:
+        settings.DEBUG_QUIETLY = True
+        q = Quietly(FakeAction())
+        assert isinstance(q, FakeAction)
+
+    def test_debug_no_kink(self, Tester: Actor, mocker: MockerFixture) -> None:
+        mocker.patch.object(settings, "DEBUG_QUIETLY", return_value=True)
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Quietly(FakeAction()).perform_as(Tester)
+        assert mock_kink.call_count == 0
+        assert mock_clear.call_count == 0
+        assert mock_flush.call_count == 0
+
+    def test_debug_logging(self, Tester: Actor, mocker: MockerFixture) -> None:
+        mocker.patch.object(settings, "DEBUG_QUIETLY", return_value=True)
+        mock_clear = mocker.spy(the_narrator, "clear_backup")
+        mock_flush = mocker.spy(the_narrator, "flush_backup")
+        mock_kink = mocker.spy(the_narrator, "mic_cable_kinked")
+
+        Tester.will(QuietlyPerformable(FakeAction()))
+        assert mock_kink.call_count == 1
+        assert mock_clear.call_count == 1
+        assert mock_flush.call_count == 1
+
+    def test_realtime_1(self, Tester, caplog) -> None:
+        caplog.set_level(logging.INFO)
+        Tester.will(Quietly(Action1()))
+        assert [r.msg for r in caplog.records] == [
+            "Tester tries to Action1",
+            "    Tester tries to Action2",
+            "        Tester sees if simpleQuestion is equal to True.",
+            "            Tester examines SimpleQuestion",
+            "                => True",
+            "            ... hoping it's equal to True.",
+            "                => <True>",
+        ]
+
+    def test_realtime_2(self, Tester, caplog) -> None:
+        caplog.set_level(logging.INFO)
+        Tester.will(Action3())
+        assert [r.msg for r in caplog.records] == [
+            "Tester tries to Action3"
+        ]
+
+    def test_realtime_3(self, Tester, caplog) -> None:
```

Comment:
Between tests realtime 1 2 and 3, what are the different things being tested?

Reply:
I probably should put this in the docstrings.

The variations are the placement of where Quietly occurs.  

- variation 1 - Quietly wrapped around action1 which inside is a quietly wrapped around action2, where inside action2 is a call to set the DEBUG flag to TRUE.  Thereby showing that WHILE we're inside of a potential kink setting the flag to log everything occurs as expected.
- variation 2 - This is basically the happy path case.  Inside action3 Quietly wraps around Action4 so we only should see the beat message from action3.
- variation 3 - quietly wraps around action2 (which again has the call to set the DEBUG flag to True) but we set the flag BEFORE making the call.  It's nearly identical to variation 1 except for where the flag is set.

Reply:
I moved tests around and renamed them. technically speaking variation 2 wasn't a test of `UNABRIDGED_LOGGING` it was proving Quietly works normally.

Reply:
Much better names and placement, thank you!